core/src/main/java/com/linecorp/armeria/server/http/auth/HttpAuthService.java

```diff
@@ -91,10 +92,20 @@ protected HttpResponse onFailure(ServiceRequestContext ctx, HttpRequest req) thr
 
     @Override
     public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-        if (authorize(req.headers())) {
-            return onSuccess(ctx, req);
-        } else {
-            return onFailure(ctx, req);
-        }
+        DeferredHttpResponse res = new DeferredHttpResponse();
+
+        authorize(req, ctx).whenComplete((result, t) -> {
```

Comment:
I remember you mentioning this once before in logging code but forgot - why would we use ```handle``` here? ```handle``` seems to be the exception-catching equivalent of ```thenApply``` - it takes a function that must return and is used for transformation. ```whenComplete``` seems to be the exception-catching equivalent of ```thenAccept``` - it takes a consumer that does not return.

In this case, I don't want to return anything since it's not a transformation, just a callback.